resources/eventing/charts/nats/values.yaml

```diff
@@ -115,7 +115,7 @@ nats:
     #                          #
     ############################
     fileStorage:
-      enabled: false
+      enabled: true
```

Comment:
Unless this is enabled, setting [this](https://github.com/kyma-project/kyma/blob/fff53022d89875a3539677120bcc4dd9702a299c/resources/eventing/charts/controller/values.yaml#L82) to 'file' does not do anything.

Reply:
I'm not so sure about that! What do you mean "doesn't do anything"? It configures the stream correctly, right? That's what it is supposed to do.

Switching the value you linked to, is for using memory or file storage for the stream (configuring a JetStream stream). When you set the value to `file` it will write to a file in `/data` (I think). The question is, what is mounted at `/data` ? If nothing, then you are writing to a file, but in the container, which means an ephemeral layer, which will be gone after a restart. 

So, yes, if you want to write to a file on a PV, then you should set that value to true. I think, it is just a matter of documentation. If you check the evaluation vs. production profile, these flagged are set. 

By permanently setting that flag to true, you will always create a PVC which would use a PV, even if you are using in-memory stream!

Reply:
The streams are created correctly but are not persisted after the restart, since we are writing to a file in the container which is ephemeral. I assume the point of using filestorage is to persist streams and hence we should created a PVC.

The question is should we set this value globally, so that when we use want streams (in the controller) to be created in memory, we do not create this PVC either?

Reply:
On second thought, according to [this line](https://github.com/kyma-project/kyma/blob/9d03ae8cbb67ebab5202ae327294a43c99831b4d/resources/eventing/charts/nats/templates/configmap.yaml#L46), I'm not sure if the storage directory is even created unless the value is set to true.

Reply:
After discussing with the team, I've moved the value to enable the storage at the global level. This would create the PVC only when we enable file storage for streams.